content/en/blog/low-hanging-performance/index.md

```diff
@@ -0,0 +1,460 @@
+specific ways. Two of my favorite 'things' At the *code level* are **readability**
+and **modifiability**. The latter is a very old friend of SW architecture's
+quality attributes ([Software Architecture in Practice, Len Bass, et
+al.](https://www.oreilly.com/library/view/software-architecture-in/9780132942799/)).
+Still, it is not well aligned with the current *practices and tools* in the OSS scene
+because everything is so text-centric. So, let's make it as readable as we can.
+
+> [*"A picture is worth a thousand words"*](https://en.wikipedia.org/wiki/A_picture_is_worth_a_thousand_words)
+
+Practise has taught that software architecture must be expressed in various
+notations; most are visual. For example, most of us can reason that a
+well-grafted state-machine diagram must be faster than a code written our
+favorite programming language. For instance, the next state-machine diagram’s
+protocol implementation is constructed of thousands of lines of code structured
+in multiple files that depend on external modules and libraries. We need
+abstraction layers to manage all that complexity.
+
+```plantuml
+@startuml
+
+title Protocol State Machine
+
+state "Waiting Start Cmd/Msg" as wstart
+
+[*] -> wstart
+wstart --> PSM: InvitationCmd
+wstart --> PSM: ConnReqMsg
+
+state "ConnectionProtocolRunning" as PSM {
+
+	state "Sending ConnReq" as sreq
+	state "Waiting ConnResp" as wresp
+
+	'[*] -left-> sreq: InvitationCmd
+	[*] --> sreq: InvitationCmd
+
+	sreq --> wresp : http/200
+	sreq: do/send until http/200
+	
+	wresp: exit/return http 200
+
+	wresp --> [*]: ConnRespMsg
+
+||
+
+	state "Sending ConnResp" as sresp
+
+	[*] --> sresp: ConnReqMsg
+
+	sresp: do/send until http 200
+
+	sresp --> [*]: http/200
+}
+
+PSM --> [*]
+
+@enduml
+```
+
+Expressing things with the control-flow structures of imperative (or functional)
+programming languages is more challenging – especially when the correctness of
+the design should be verified. It seems that it’s easy to forget software
+quality attributes during fast-phasing programming if we use tools that only
+manage sequential text, i.e., code. At the code level, we should use functions
+that give us an abstraction hierarchy and help us to maintain readability.
+
+Moreover, since my studies of SW architecture’s quality attributes, I have
+understood that modifiability is more than modularity, re-usability, or using
+correct architectural styles like pipe and filter. Now we understand the importance of
+[TTD](https://en.wikipedia.org/wiki/Test-driven_development), [continuous
+deployment](https://en.wikipedia.org/wiki/Continuous_deployment),
+[DevOps](https://en.wikipedia.org/wiki/DevOps), etc. The practices above
+don't work only on *one* of the engineering domains. The best results are
+achieved **cross-domain engineering practices**. 
+
+In this post, I'll go through some fundamental Go tools and tricks you could use
+to achieve *decent execution speed of your (Go) software and maintain or
```

Comment:
It feels you are trying to cover both readability and performance in this text. It feels like the topics would deserve a dedicated blog each.